lib/adapters/code-action-adapter.ts

```diff
@@ -81,34 +82,50 @@ export default class CodeActionAdapter {
 
   private static async executeCommand(command: any, connection: LanguageClientConnection): Promise<void> {
     if (Command.is(command)) {
-      await CommandExecutionAdapter.executeCommand(connection, command.command, command.arguments)
+      await connection.executeCommand({
+        command: command.command,
+        arguments: command.arguments,
+      })
     }
   }
+}
 
-  private static createCodeActionParams(
+function createCodeActionParams(
     linterAdapter: LinterPushV2Adapter,
     editor: TextEditor,
     range: Range,
     diagnostics: atomIde.Diagnostic[]
```

Comment:
I think we need to break the old `atomIde.Diagnostic` in favor of `Linter.Message` here.

Reply:
No. We can provide service for both in AutoLanguageClient. We just need to call the correct function.